dimod/reference/composites/scalecomposite.py

```diff
@@ -0,0 +1,284 @@
+       ...                ignored_interactions=[('a','b')],num_reads = 100)
+       >>> print(response.aggregate().record) # doctest: +SKIP
+       >>> response = sampler.sample_ising(linear, quadratic, scalar=0.5,
+       ...                                              num_reads = 100)
+       >>> print(response.aggregate().record) # doctest: +SKIP
+       [([-1,  1], -3.2, 58) ([ 1, -1], -3.2, 42)]
+       [([1, 1], -4.8, 100)]
+
+    """
+
+    def __init__(self, child_sampler):
+        self._children = [child_sampler]
+
+    @property
+    def children(self):
+        return self._children
+
+    @property
+    def parameters(self):
+        return self.child.parameters.copy()
+
+    @property
+    def properties(self):
+        return {'child_properties': self.child.properties.copy()}
+
+    def sample(self, bqm, scalar=None, bias_range=1, quadratic_range=None,
+               ignored_variables=None, ignored_interactions=None,
+               ignore_offset=False, **parameters):
+        """ Scale and sample from the provided binary quadratic model.
+
+        if scalar is not given, problem is scaled based on bias and quadratic
+        ranges. See :meth:`.BinaryQuadraticModel.scale` and
+        :meth:`.BinaryQuadraticModel.normalize`
+
+        Args:
+            bqm (:obj:`dimod.BinaryQuadraticModel`):
+                Binary quadratic model to be sampled from.
+
+            scalar (number):
+                Value by which to scale the energy range of the binary quadratic model.
+
+            bias_range (number/pair):
+                Value/range by which to normalize the all the biases, or if
+                `quadratic_range` is provided, just the linear biases.
+
+            quadratic_range (number/pair):
+                Value/range by which to normalize the quadratic biases.
+
+            ignored_variables (iterable, optional):
+                Biases associated with these variables are not scaled.
+
+            ignored_interactions (iterable[tuple], optional):
+                As an iterable of 2-tuples. Biases associated with these interactions are not scaled.
+
+            ignore_offset (bool, default=False):
+                If True, the offset is not scaled.
+
+            **parameters:
+                Parameters for the sampling method, specified by the child sampler.
+
+        Returns:
+            :obj:`dimod.SampleSet`
+
+        """
+
+        child = self.child
+
+        bqm_copy = bqm.copy()
+        if scalar is None:
+            bqm_copy.normalize(bias_range, quadratic_range,
+                               ignored_variables=ignored_variables,
+                               ignored_interactions=ignored_interactions,
+                               ignore_offset=ignore_offset)
+        else:
+            bqm_copy.scale(scalar, ignored_variables=ignored_variables,
+                           ignored_interactions=ignored_interactions,
+                           ignore_offset=ignore_offset)
```

Comment:
In this case you don't need to recalculate the energies, you can just scale them

Reply:
I can't really just rescale them. there are options to ignore certain variables and interactions.